lib/mapit_api.rb

```diff
@@ -1,9 +1,13 @@
 module MapitApi
   class InvalidPostcodeError < StandardError; end
+  class ValidPostcodeNoPlaces < StandardError; end
 
   def self.location_for_postcode(postcode)
     location_data = Imminence.mapit_api.location_for_postcode(postcode)
     raise InvalidPostcodeError if location_data.nil?
+    if areas = location_data.response.to_hash["areas"]
```

Comment:
The `.response.to_hash["areas"]` feels redundant as I think you could just use `location_data.areas` here (apols. if you tried this already and it worked, or you had some other reason for doing it this way)

However, I think checking for the presence of some `areas` is the wrong thing to look for to guard against the problem we have.  It's the `lat` / `lon` attributes we'll try to use to construct the `Point` later on and those being `nil` is what causes the problem, so we should be checking for them, not `areas`.  It's also worth noting that we could have a response from mapit for a postcode that has no `areas` but has `lat`/`lon`, or has no `lat`/`lon` but does have some `areas`.